File: plldb-server/src/main/java/cn/twincest/plldb/util/Result.java
package cn.twincest.plldb.util;

import com.google.gson.JsonObject;

public class Result {

	public String status;

	public JsonObject data;

	public Result() {
		this.status = null;
		this.data = new JsonObject();
	}

	public <T> void append(String key, T value) {
		this.data.add(key, JSONUtil.toTree(value));
	}

	public JsonObject generate() {
		JsonObject result = new JsonObject();
		if (this.status != null) {
			result.addProperty("status", this.status);
		} else {
			result.add("data", JSONUtil.toTree(data));
		}
		return result;
	}

}

File: plldb-server/src-refactor/main/java/cn/twin_kles/plldb/util/Result.kt
package cn.twin_kles.plldb.util

import com.google.gson.JsonObject

class Result {

	var status: String? = null
    var data: JsonObject = JsonObject()

    fun <T> append(key: String, value: T) {
        data.add(key, JSONUtil.toTree(value))
    }

    fun generate(): JsonObject {
        val result = JsonObject()
        if (status != null) {
            result.addProperty("status", status)
        } else {
            result.add("data", JSONUtil.toTree(data))
        }
        return result
    }

}